third_party/txt/src/txt/paragraph.cc

```diff
@@ -1458,12 +1458,17 @@ std::vector<Paragraph::TextBox> Paragraph::GetRectsForRange(
                            box.direction);
       }
     } else if (rect_height_style == RectHeightStyle::kStrut) {
-      for (const Paragraph::TextBox& box : kv.second.boxes) {
-        boxes.emplace_back(
-            SkRect::MakeLTRB(
-                box.rect.fLeft, line_baselines_[kv.first] - strut_.ascent,
-                box.rect.fRight, line_baselines_[kv.first] + strut_.descent),
-            box.direction);
+      if (strut_.force_strut) {
```

Comment:
Try `paragraph_style_.strut_enabled && paragraph_style_.strut_font_size >= 0` as the condition?

Reply:
Updated this to use the same criteria as `valid_strut` in `ComputeStrut`